pkg/scheduler/objects/application.go

```diff
@@ -126,8 +126,8 @@ type Application struct {
 	appEvents             *schedEvt.ApplicationEvents
 	sendStateChangeEvents bool // whether to send state-change events or not (simplifies testing)
 
-	snapshot        bytes.Buffer
-	appSnapshotLock locking.RWMutex
+	snapshot     bytes.Buffer
```

Comment:
In general I'm in favor of better naming like this, but does the risk of creating future conflicts with the upstream fork outweigh it? I could be convinced either way, but it was my first thought seeing this....

Reply:
I don't think it will be an issue - each `Application`, `Queue`, and `Node` have the single-object-scoped `snapshot` (basically holds a point-in-time JSON dump of the object) and the mutex lock was added to each (also single-object-scoped) to fix race/deadlock issues that were happening in the YK core unit tests. This name-shortening is mostly to prevent "name-stuttering" when calling the method on an object, e.g. many places `Application` objects are just named `app`, so this change avoids having `app.appSnapshotLock.Lock()` and it becomes just `app.snapshotLock.Lock()`.   (Maybe calling it `app.snapshotMutex` would be better, so we don't have two "Lock"s together?)